Sources/Sentry/Profiling/SentryProfiledTracerConcurrency.mm

```diff
@@ -49,9 +73,46 @@
     }
 }
 
-} // namespace
+/**
+ * Decrement number of root spans in flight and stop continuous profiler if there are none left.
+ * @warning Must be called from a synchronized context.
+ */
+void
+_unsafe_cleanUpContinuousTraceProfiler()
+{
+    if (SENTRY_CASSERT_RETURN(_gInFlightRootSpans > 0,
+            @"Attempted to decrement count of root spans to less than zero.")) {
+        _gInFlightRootSpans -= 1;
+    }
```

Comment:
`m`: I'm unsure if we shouldn't choose an approach that degrades gracefully here. If we have a bug in our logic, this would make the SDK unusable in debug builds. Wouldn't it be better to only log a error or fatal log and continue? This applies to all SENTRY_CASSERT_RETURN. As this involves span tracking and async code is involved, there is a risk for having such a bug.

Reply:
My current thinking on this is that I want to know about such an unbalancing, should it pop up in unit tests. I don't think it currently is possible, and can't think of a way to write a test for it, so I left this here to point it out should it somehow occur in the future. 

I guess I look at this as an inline unit test, but I did want it to degrade gracefully for customers, however I see what you're saying about debug builds. How about I conditionally compile in an assertion only for SENTRY_TEST and SENTRY_TESTCI configs?

Reply:
You could assert the log output of your tests. So, basically, if the specific log message ever gets logged, let the test fail. But that won't fail then if some other future test ends up in this part of the code.

Using `SENTRY_CASSERT_RETURN` because we don't know how to test a specific part of the code, seems like a code smell to me and maybe we should restructure our code so we can test it. I would prefer having extracting the logic so we can test various scenarios and don't have to rely on `SENTRY_CASSERT_RETURN` to potentially reveil future problems. I can't tell you the exact solution right now. If you want I can take a closer look an open a draft PR with my some ideas, @armcknight.

Reply:
I see what you're saying, and I'll think a little bit more about how I can exercise the codepaths that traverse these lines. 

But there are reasonable uses for asserts. Unit tests ensure proper operation of the codepaths as they exist along with the tests. Assertions ensure proper operation of a particular part of a codepath _when the code is being changed_–you can't write all the unit tests exercising all the possible future code paths that could ever exist when this code is touched later, by anyone.

Reply:
Both approaches have pros and cons, as always 😃. In this specific case, I think it's OK to keep the assertions.